apps/explorer/priv/repo/migrations/20220617083341_set_counters_on_transaction_table.exs

```diff
@@ -0,0 +1,61 @@
+defmodule Explorer.Repo.Migrations.SetCountersOnTransactionTable do
+  use Ecto.Migration
+
+  def up do
+    execute("CREATE TABLE celo_transaction_stats(stat_type varchar(255), value numeric(100,0));")
+
+    # coalesce gas_used values to prevent null propogation when gas used is null
+    execute("""
+    CREATE OR REPLACE FUNCTION celo_transaction_stats_trigger_func() RETURNS trigger
+    LANGUAGE plpgsql AS
+    $$BEGIN
+    IF TG_OP = 'INSERT' THEN
+      UPDATE celo_transaction_stats SET value = value + 1 WHERE stat_type = 'total_transaction_count';
+      UPDATE celo_transaction_stats SET value = value + coalesce(NEW.gas_used, 0) WHERE stat_type = 'total_gas_used';
+
+      RETURN NEW;
+    ELSIF TG_OP = 'DELETE' THEN
+      UPDATE celo_transaction_stats SET value = value - 1 WHERE stat_type = 'total_transaction_count';
+      UPDATE celo_transaction_stats SET value = value - coalesce(OLD.gas_used, 0) WHERE stat_type = 'total_gas_used';
+
+      RETURN OLD;
+    ELSIF TG_OP = 'UPDATE' THEN
+      UPDATE celo_transaction_stats SET value = value + (coalesce(NEW.gas_used, 0) - coalesce(OLD.gas_used, 0)) WHERE stat_type = 'total_gas_used';
+
+      RETURN NEW;
+    ELSIF TG_OP = 'TRUNCATE' THEN
+      UPDATE celo_transaction_stats SET value = 0 WHERE stat_type = 'total_transaction_count';
+      UPDATE celo_transaction_stats SET value = 0 WHERE stat_type = 'total_gas_used';
+
+      RETURN NULL;
+    END IF;
+    END;$$;
+    """)
+
+    execute("""
+    CREATE CONSTRAINT TRIGGER celo_transaction_stats_modified
```

Comment:
Ah, sorry, missed `OR REPLACE` here and below.

Reply:
https://github.com/celo-org/blockscout/runs/7004593995?check_suite_focus=true#step:10:1593

```
13:07:52.744 [info]  execute "CREATE OR REPLACE CONSTRAINT TRIGGER celo_transaction_stats_modified\nAFTER INSERT OR DELETE OR UPDATE ON transactions\nDEFERRABLE INITIALLY DEFERRED\nFOR EACH ROW EXECUTE PROCEDURE celo_transaction_stats_trigger_func();\n"

** (Postgrex.Error) ERROR 0A000 (feature_not_supported) CREATE OR REPLACE CONSTRAINT TRIGGER is not supported
```

It seems like this syntax doesn't exist for triggers

Reply:
https://stackoverflow.com/questions/35927365/create-or-replace-trigger-postgres#comment77864054_39650162

Reply:
https://www.postgresql.org/docs/current/sql-createtrigger.html it's not supported by our postgres version it seems